core/trino-main/src/test/java/io/trino/sql/planner/TestUnwrapCastInComparison.java

```diff
@@ -811,12 +819,21 @@ private void testUnwrap(String inputType, String inputPredicate, String expected
         testUnwrap(getQueryRunner().getDefaultSession(), inputType, inputPredicate, expectedPredicate);
     }
 
+    private void testUnwrap(String inputType, String inputPredicate, String expectedPredicate, boolean reverseOrder)
```

Comment:
What's `reverseOrder` and why do we need it?

Reply:
The usage of that specific query and predicate is an internal implementation detail of the `testUnwrap` function. If that's not working in general, we need to come up with a different way to test that an expression is being unwrapped properly.

Regarding the expression being reordered by the engine, we should look into why that is happening. In general, we try to preserve the order unless we can determine that one expression is cheaper to evaluate first (they currently get evaluated left to right)